File: library/src/test/java/me/proxer/library/enums/AnimeLanguageTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class AnimeLanguageTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final AnimeLanguage animeLanguage = api.moshi().adapter(AnimeLanguage.class).fromJson("\"gersub\"");

        assertThat(animeLanguage).isSameAs(AnimeLanguage.GERMAN_SUB);
    }

    @Test
    void testFallback() throws IOException {
        final AnimeLanguage animeLanguage = api.moshi().adapter(AnimeLanguage.class).fromJson("\"xyz\"");

        assertThat(animeLanguage).isSameAs(AnimeLanguage.OTHER);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/AnimeLanguageTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class AnimeLanguageTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(AnimeLanguage::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"gersub\"") shouldBe AnimeLanguage.GERMAN_SUB
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe AnimeLanguage.OTHER
    }
}
